Consolidate Metaflow infra tooling - initial pass (#13)
outerbounds/metaflow-tools: README.md

 
# ⚒️ Metaflow Admin Tools

This repository contains various configuration files, tools and utilities for operating [Metaflow](https://github.com/Netflix/metaflow) in production. See [Metaflow documentation](https://docs.metaflow.org) for more information about Metaflow architecture. Top level folders are structured as follows:

## Metaflow on AWS (/aws)
### Metaflow Cloudformation template
If you're not already using Terraform, this is the easiest way to get started on AWS. You can find the template under [aws/cloudformation](./aws/cloudformation) in this repository.

### Sample Terraform templates for AWS

> An alternate infrastructure as code option is Terraform. An end-to-end example can be found under [aws/terraform](./aws/terraform) in this repository. The example leverages the official Terraform module [outerbounds/terraform-aws-metaflow](https://registry.terraform.io/modules/outerbounds/metaflow/aws/latest) as a building block.

Reviewer: Purely phrasing nit: I found the first sentence kinda hard to parse, had to re-read it a couple times ("an alternate infrastructure.. what ?"). Maybe its just me, but something like this would be easier for me:

```suggestion
Another deployment option is using Terraform. An end-to-end example can be found under [aws/terraform](./aws/terraform) in this repository. The example leverages the official Terraform module [outerbounds/terraform-aws-metaflow](https://registry.terraform.io/modules/outerbounds/metaflow/aws/latest) as a building block.
```

## Metaflow on Azure (/azure)
### Sample Terraform templates for Azure
This is the quickest way to spin up a fully functional Metaflow stack on Azure. See details under [azure/terraform](./azure/terraform) in this repository.

## Metaflow services on Kubernetes (/k8s)
### Helm Charts (alpha)
We provide Helm charts to deploy Metaflow Metadata service and UI in a K8S cluster. This way you can use Metaflow without any AWS-specific dependencies on AWS except for having a S3-compatible object storage engine available. You can find them under [k8s/helm/metaflow](./k8s/helm/metaflow) in this repository.

> These Helm charts are NOT needed if you are using Azure sample templates above - those templates are all-inclusive.

Reviewer: ```suggestion
```
I dont' think you need to qualify it here. Reading this README I wouldn't assume that they are somehow related to Azure in the first place

## Cloud agnostic resources (/common)
### Sample Metaflow flow definitions

> Some Metaflow flows that could be used to test drive a Metaflow stack (e.g. running in a cloud). Some of these flows

Reviewer: Another phrasing nit: IMO it may not be clear to the reader what does "test drive .. e.g. running in a cloud" means

I'd either remove the parens
```suggestion
Some Metaflow flows that could be used to test drive a Metaflow stack. Some of these flows
```

Or clarify
```suggestion
Some Metaflow flows that could be used to confirm that Metaflow stack works after deploying it in a cloud environment. Some of these flows
```
